src/App.tsx

```diff
@@ -1,27 +1,79 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const enum Action {
+  AllGoods = 'all-goods',
+  FirstFive = 'first-five',
+  OnlyRed = 'red',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+  const [handleAction, setHandleAction] = useState(Action.AllGoods);
```

Comment:
the `handleAction` state variable isn't necessary here. you can just call the function on click and set visibleGoods